src/main/scala/main.scala

```diff
@@ -54,4 +52,27 @@ object RockSlicer {
     jsonBlocks.saveAsTextFile("blocks.json")
     sc.stop()
   }
+
+  // Produce joints that achieve some lead balancing when generating the initial Block RDD
+  private def generateSeedJoints(joints: Seq[Joint], numSeedJoints: Integer): (Seq[Joint], Seq[Joint]) = {
+    val (persistentJoints, nonPersistentJoints) = joints.partition(_.shape.isEmpty)
+    persistentJoints.length match {
+      // All of the seed joints will be persistent
+      case l if l >= numSeedJoints => (persistentJoints.take(numSeedJoints),
```

Comment:
The variable  `l` name is a bit confusing - looks like a one. Maybe use something else?